app/src/components/market/sections/market_create/steps/items/create_scalar_market.tsx

```diff
@@ -125,6 +120,62 @@ export const CreateScalarMarket = (props: Props) => {
     upperBound,
   } = props
 
+  let lowerBoundError
+  let startingPointError
+  let upperBoundError
+  let startingPointNumber
+  let lowerBoundNumber
+  let upperBoundNumber
+  // if the input values aren't null the number is cast to BigNumber with 18 decimals and precision value of 2
+  if (startingPoint !== null) {
+    startingPointNumber = formatBigNumber(startingPoint, 18, 2)
+  }
+
+  if (lowerBound !== null) {
+    lowerBoundNumber = formatBigNumber(lowerBound, 18, 2)
+  }
+
+  if (upperBound !== null) {
+    upperBoundNumber = formatBigNumber(upperBound, 18, 2)
+  }
+  //check for values to be non negative integers
+  if (startingPointNumber !== undefined && startingPointNumber < '0') {
+    startingPointError = 'Value must be greater than 0'
+  }
+
+  if (lowerBoundNumber !== undefined && lowerBoundNumber < '0') {
+    lowerBoundError = 'Value must be greater than 0'
+  }
+
+  if (upperBoundNumber !== undefined && upperBoundNumber < '0') {
+    upperBoundError = 'Value must be greater than 0'
+  }
+  //checks if error is appropriate or not. if true, displays error and removes decimals from numbers in error message.
+  //checks for starting point to not be undefined and not equal 0. this prevents unwanted error messages like "lower bound must be less than 0."
+  if (
+    Number(startingPointNumber) > Number(upperBoundNumber) &&
```

Comment:
I would convert the values to numbers right away upon assigning them as it would better reflect the variable names and work better with comparison checks

Reply:
i had it like this originally but roy told me that casting big number type to Number was dangerous